File: POO_Exemplos/CPP/poo.cpp
#include <iostream>
using namespace std;

// Abstração + Encapsulamento
class Conta {
private:
    double saldo;
public:
    Conta() : saldo(0) {}

    void depositar(double valor) {
        if (valor > 0)
            saldo += valor;
    }

    double getSaldo() {
        return saldo;
    }
};

// Herança + Polimorfismo
class Animal {
public:
    virtual void emitirSom() {
        cout << "Som genérico" << endl;
    }
};

class Cachorro : public Animal {
public:
    void emitirSom() override {
        cout << "Latido" << endl;
    }
};

File: POO_Exemplos/Python/poo.py
# Abstração + Encapsulamento
class Conta:
    def __init__(self):
        self.__saldo = 0  # atributo privado

    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor

    def get_saldo(self):
        return self.__saldo

# Herança + Polimorfismo
class Animal:
    def emitir_som(self):
        print("Som genérico")

class Cachorro(Animal):
    def emitir_som(self):
        print("Latido")